src/components/KTruncate/KTruncate.vue

```diff
@@ -0,0 +1,280 @@
+     */
+    showToggle.value = containerHeightProperty > (kTruncateWrapper.value.offsetHeight - textToggleControlsHeight)
+    countExcessElements()
+  }
+}
+
+// Counts elements that are wrapped to the hidden rows and therefore are not visible.
+const countExcessElements = (): void => {
+  if (props.isTextContent) {
+    return
+  }
+
+  excessElementsCount.value = 0
+  if (kTruncateContainer.value && kTruncateWrapper.value) {
+    const children = kTruncateContainer.value.children as unknown as HTMLElement[]
+    for (let i = 0; i < children.length; i++) {
+      /**
+       * If element's offsetTop
+       * (offset from the nearest relatively positioned parent, which is .k-truncate-container)
+       * is greater than the wrapper element height - means it's not visible
+       */
+      if (children[i].offsetTop > kTruncateWrapper.value.offsetHeight) {
+        excessElementsCount.value += 1
+      }
+    }
+  }
+}
+
+const handleToggleClick = async () => {
+  expanded.value = !expanded.value
+  // await for component to collapse/expand
+  await nextTick()
+  updateToggleVisibility()
+}
+
+onMounted(() => {
+  resizeObserver.value = new ResizeObserver(updateToggleVisibility).observe(kTruncateContainer.value as HTMLDivElement)
+})
+
+onBeforeUnmount(() => {
+  if (resizeObserver.value) {
+    resizeObserver.value.unobserve(kTruncateContainer.value)
+  }
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+
+.k-truncate {
+  align-items: flex-start;
+  display: flex;
+  overflow: hidden;
+  width: 100%;
+
+  .expand-trigger, .collapse-trigger {
+    --KButtonBtnLink: var(--KTruncateToggleColor, var(--blue-500, color(blue-500)));
+    --KButtonPrimaryBase: var(--KTruncateCollapseBackground, var(--blue-100, color(blue-100)));
+    --KButtonPrimaryHover: var(--KTruncateCollapseHover, var(--blue-200, color(blue-200)));
+    --KButtonPrimaryActive: var(--KTruncateCollapseHover, var(--blue-200, color(blue-200)));
+  }
+
+  &.k-truncate-content {
+    height: v-bind('wrapperHeight');
+
+    &.expanded {
+      height: auto;
+    }
+    .k-truncate-container {
+      display: flex;
+      flex-wrap: wrap;
+      gap: 10px;
+      position: relative;
+    }
+
+    .expand-trigger {
+      font-size: var(--type-xs);
+
+      &::before {
+        content: '+';
+        margin-right: -7px;
+      }
+    }
+    .collapse-trigger {
+      padding: var(--spacing-xxs);
+
+      &:focus, &:active {
+        box-shadow: none;
+      }
+    }
+  }
+  &.k-truncate-text {
+    display: flex;
+    flex-direction: column;
+
+    .k-truncate-container {
+      -webkit-box-orient: vertical;
+      display: -webkit-box;
+      -webkit-line-clamp: v-bind('props.rows');
+      overflow: hidden;
+    }
+    &.expanded {
+      .k-truncate-container {
+        display: block;
+      }
+    }
+    .expand-trigger, .collapse-trigger {
```

Comment:
nit: move to new line
```suggestion
    .expand-trigger, 
    .collapse-trigger {
```